File: service_invocation/java/OrderProcessingService/src/main/java/com/service/OrderProcessingService/controller/OrderProcessingServiceController.java
package com.service.OrderProcessingService.controller;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.client.RestTemplate;

import java.util.Random;
import java.util.concurrent.TimeUnit;

@RestController
@RequestMapping("/")
public class OrderProcessingServiceController {

    private static final Logger log = LoggerFactory.getLogger(OrderProcessingServiceController.class);

    @GetMapping(path="/order", produces = "application/json")
    public void getOrder() {
        Random random = new Random();
        int orderId = random.nextInt(1000-1) + 1;
        String daprPort = "3602";
        String daprUrl = "http://localhost:" + daprPort + "/v1.0/invoke/checkout/method/checkout/" + orderId;
        RestTemplate restTemplate = new RestTemplate();
        String result = restTemplate.getForObject(daprUrl, String.class);
        log.info("Order requested: " + orderId);
        log.info("Result: " + result);
    }
}

File: service_invocation/csharp/OrderProcessingService/Controllers/OrderProcessingServiceController.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using System;
using Microsoft.AspNetCore.Mvc;
using System.Net.Http;
using System.Net.Http.Headers; 

namespace OrderProcessingService.controller
{
    [ApiController]
    [Route("order")]
    public class OrderProcessingServiceController : Controller
    {
        [HttpGet]
        public void getOrder()
        {
            System.Threading.Thread.Sleep(5000);
            Random random = new Random();
            int orderId = random.Next(1,1000);
            string daprPort = "3602";
            string daprUrl = "http://localhost:" + daprPort + "/v1.0/invoke/checkout/method/checkout/" + orderId;
            string urlParameters = "";
            HttpClient client = new HttpClient();
            client.BaseAddress = new Uri(daprUrl);
            client.DefaultRequestHeaders.Accept.Add(
            new MediaTypeWithQualityHeaderValue("application/json"));
            HttpResponseMessage result = client.GetAsync(urlParameters).Result;
            Console.WriteLine("Order requested: " + orderId);
            Console.WriteLine("Result: " + result);
        }
    }
}